core/key.go

```diff
@@ -283,6 +307,30 @@ func (k CollectionIndexKey) ToDS() ds.Key {
 	return ds.NewKey(k.ToString())
 }
 
+func NewSchemaHistoryKey(schemaId string, previousSchemaVersionID string) SchemaHistoryKey {
+	return SchemaHistoryKey{
+		SchemaID:                schemaId,
+		PreviousSchemaVersionID: previousSchemaVersionID,
+	}
+}
+
+func NewSchemaVersionMigrationKey(schemaVersionID string) SchemaVersionMigrationKey {
+	return SchemaVersionMigrationKey{SourceSchemaVersionID: schemaVersionID}
+}
+
+func NewSchemaHistoryKeyFromString(keyString string) (SchemaHistoryKey, error) {
+	keyString = strings.TrimPrefix(keyString, COLLECTION_SCHEMA_VERSION_HISTORY+"/")
+	elements := strings.Split(keyString, "/")
+	if len(elements) != 2 {
+		return SchemaHistoryKey{}, ErrInvalidKey
```

Comment:
todo: this one is not covered with tests although it's very easy to add in `key_test.go`

Reply:
What makes this dead code? Its a safety check data that came from an unstructured string manipulation, itll idealy save dev time in the future if someone accidentally misuses the `SchemaHistoryKeyFromString` func. Feels fairly necessary/helpful to me, otherwise the code will panic below.

Reply:
> otherwise the code will panic below

Only if the database has been corrupted.  There is no sensible way to reach this code via public entry points (technically you can by messing with the `DB.RootStore`, but I class that as corrupting the Db).

Reply:
I mean an internal dev making changes, it could be fairly easy to accidentally fuck up, we're always trying to protect users and future maintainers. I dont have a super strong opinion regarding the testing, but I do have a strong opinion that this isnt dead code and adds safety for a myriad of reasons.

Reply:
I like to think a myriad of other tests would fail if an internal dev managed to mess this up.

I've added an issue to test this properly: https://github.com/sourcenetwork/defradb/issues/1629